src/licensedcode/models.py

```diff
@@ -825,7 +825,7 @@ class BasicRule(object):
     # for SPDX license expression dynamic rules or testing
     stored_text = attr.ib(default=None, repr=False)
 
-    key_phrase_spans = attr.ib(default=None, repr=False)
+    key_phrase_spans = attr.ib(default=[], repr=False)
```

Comment:
You cannot use a list instance as a default, that's a global with improbable behaviour. Instead use `attr.Factory(list)` which will lazily construct a new default instance for each new object.
 ```suggestion
    key_phrase_spans = attr.ib(default=attr.Factory(list), repr=False)
```